src/rtapi/rtapi_compat.c

```diff
@@ -116,27 +116,32 @@ int xenomai_gid()
 int user_in_xenomai_group()
 {
     int numgroups, i;
-    gid_t *grouplist;
+    gid_t *grouplist = NULL;
```

Comment:
I disagree, you could say the same thing about the 
int i; declaration.
I hate local variables being buried inside local scope conditionals, it makes the code very hard to read.
This amendment just ensures the variable is NULL, which it wasn't originally set.
Just belt and braces as it is explicitly tested now